File: script/client-simulation.ts
import OpenAI from "https://deno.land/x/openai@v4.67.3/mod.ts";

const openai = new OpenAI({ 
    apiKey: "ollama", 
    baseURL: "http://localhost:8000/v1" ,
    defaultHeaders: {
        "x-openai-base-url": "http://localhost:11434/v1"
    }
});

async function main() {
    const result = await openai.chat.completions.create({
        model: "qwen:4b",
        messages: [
            { role: "system", content: "You are a helpful assistant." },
            { role: "user", content: "What is the capital of the United States?" },
        ],
        stream: false,
    });
    console.log(result.choices[0].message.content);
}

main()

File: script/client-simulation.py
from openai import OpenAI

client = OpenAI(
    api_key="ollama",
    base_url="http://localhost:8000/v1",
    default_headers={ "x-openai-base-url": "http://localhost:11434/v1" }
)

chat_completion = client.chat.completions.create(
    messages=[
        {
            "role": "user",
            "content": "Say this is a test",
        }
    ],
    model="qwen:4b",
    stream=False,
)
print(chat_completion.choices[0].message.content)
